File: engine/shopkeeper.lua
local Shopkeeper = setmetatable(Worker, {})
Shopkeeper.__index = Shopkeeper

function Shopkeeper:new(x, y)
  local self = setmetatable({}, Shopkeeper)
  self:__init2(x, y)
  return self
end

function Shopkeeper:type()
  return 'Shopkeeper'
end


-------------
-- PRIVATE --
-------------

function Shopkeeper:__tostring()
  return '[Shopkeeper]'
end

return Shopkeeper

-- vim: ts=2:sw=2:sts=2:expandtab

File: engine/workers/shopkeeper.cc
#include "engine/workers/shopkeeper.h"

#include "./globals.h"
#include "ui/ui.h"

Shopkeeper::Shopkeeper(Point init, class City& city)
    : Worker(init, city)
{
}

Shopkeeper::~Shopkeeper()
{
}


void
Shopkeeper::TalkToHero()
{
    ui->Dialog().Shopkeeper(city, *this);
}

// vim: ts=4:sw=4:sts=4:expandtab
